File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    return "Whoa, chill out!" if remark.upcase == remark && remark.downcase != remark.upcase
    return "Sure." if remark.match(/[?]\z/)
    return "Fine. Be that way!" if remark =~ /\A\s*\z/
    "Whatever."
  end
end

File: go/bob/bob.go
package bob

import (
	"strings"
)

// Hey returns Bob's response to your input
func Hey(remark string) string {
	remarkRunes := []rune(strings.TrimSpace(remark))

	if strings.ToUpper(remark) == remark && strings.ToUpper(remark) != strings.ToLower(remark) {
		return "Whoa, chill out!"
	} else if len(remarkRunes) == 0 {
		return "Fine. Be that way!"
	} else if remarkRunes[len(remarkRunes)-1] == 63 {
		return "Sure."
	}
	return "Whatever."
}
